management/WFCORE-5489_override_attribute_value_from_env_var.adoc

```diff
@@ -0,0 +1,152 @@
+$ ./bin/standalone.sh
+----
+
+If he runs WildFly on the Cloud, he needs to set this *env var* on its pod container without any modificaiton to the application image.
+
+This RFE will simplify the integration of services with WildFly with information that is only known during deployment (such as URLs, credentials, etc.)
+
+== Resource and attribute mapping
+
+The mapping is straightforward. The name of the env var is based on thethe address of the resource and the name of the attribute:
+
+1. take the address of the resource
+** `/subsystem=undertow/server=default-server/http-listener=default`
+2. remove the leading slash (`/`)
+** `subsystem=undertow/server=default-server/http-listener=default`
+3. append *two underscores* (`__`) and the name of the attribute:
+** `subsystem=undertow/server=default-server/http-listener=default__proxy-address-forwarding`
+4. Replace all non-alphanumeric characters with an underscore (`_`) and put it in upper case
+** `SUBSYSTEM_UNDERTOW_SERVER_DEFAULT_SERVER_HTTP_LISTENER_DEFAULT__PROXY_ADDRESS_FORWARDING`
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5489[As a developer, I want to override management attribute values using environment variables]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFCORE-5464[WFCORE-5464 - Add environment variables as a source for model expression resolution]
+** this is a different feature as it reads environment variables *if the attribute value is an expression*. With WFCORE-5489, we override the attribute value but do not specify its semantics (it can be an expression if the attribute allows it).
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+
+* [x] Engineering
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly-core
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [x] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+Everytime an attribute value is validated and set in the management model, we look if there is a corresponding environment value that overrides it *before* doing any expression resolution or correction.
+
+This change will be done in `org.jboss.as.controller.AttributeDefinition#validateAndSet`. Any attribute definition that does not use this method to validate and set their values will not be taken into account.
+
+This feature is activated by a presence of an environment variable: `WILDFLY_OVERRIDING_ENV_VARS`.
+
+As the target of this feature is application images running on the cloud, we will make it the default behaviour for the S2I images by setting this `WILDFLY_OVERRIDING_ENV_VARS` in the builder and runtime images.
+
+For cloud deployment with Bootable Jar, the users will have to set this env var in addtion to any env var that will overrides attribute values.
```

Comment:
I think noting some items related to persistence would be good:

1) If an attribute value is determined from an environment variable, the next time the configuration is persisted, that value from the environment variable will be persisted.
2) Until something triggers persistence of the configuration file (which may never happen), the configuration file will not reflect the current running configuration.

These aren't really 'requirements', they are more a description of the intended behavior, such that doing something else is *not* a requirement.  I'll add an explicit related non-requirement below.

Reply:
I've added some items about persistence